File: dsa-learning/dsa-java-code/src/main/java/com/dsa/leetcode/array/P26_RemoveDuplicatesFromSortedArray.java
package com.dsa.leetcode.array;

public class P26_RemoveDuplicatesFromSortedArray {
    public static int removeDuplicates(int[] nums) {
        int i = 0;
        for (int j = 1, len = nums.length; j < len; j++) {
            if (nums[i] < nums[j]) {
                nums[++i] = nums[j];
            }
        }
        return i + 1;
    }

    public static void main(String[] args) {
        int[] nums = {0, 0, 1, 1, 2, 2};
        System.out.println(removeDuplicates(nums));
    }
}

File: dsa-learning/dsa-python-code/com/dsa/leetcode/array/P26_RemoveDuplicatesFromSortedArray.py
# author: tobebetter9527
# since: 2023/11/3 9:35
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        for j in range(len(nums)):
            if nums[i] < nums[j]:
                i = i + 1
                nums[i] = nums[j]
        return i + 1


if __name__ == '__main__':
    nums = [0, 0, 1, 1, 2, 2]
    solu = Solution()
    print(solu.removeDuplicates(nums))
